finetuner/tuner/pytorch/miner.py

```diff
@@ -59,6 +63,99 @@ def mine(
         diffs = matches ^ 1
 
         matches.fill_diagonal_(0)
+
+        triplets = matches.unsqueeze(2) * diffs.unsqueeze(1)
+
+        return torch.where(triplets)
+
+
+class TripletEasyHardMiner(TripletMiner):
```

Comment:
Please provide a docstring. Also, do you need to subclass `TripletMiner` - it doesn't seem to share any methods with it.